ml-core/base/src/main/java/com/stripe/android/mlcore/base/InterpreterInitializer.kt

```diff
@@ -0,0 +1,20 @@
+package com.stripe.android.mlcore.base
+
+import android.content.Context
+import androidx.annotation.RestrictTo
+
+/**
+ * Initializer for TFLite Interpreter. Needs to be run before any inference calls.
+ */
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+interface InterpreterInitializer {
+    /**
+     * Initialize the encapsulated TFLite interpreter. Needs to be run on a non-UI thread.
+     *
+     * @param context: Application context
+     * @param onSuccess: Notifies initialization success, can only run inference after this
+     * @param onFailure: Notifies initialization failure
+     */
+    suspend fun initialize(context: Context, onSuccess: () -> Unit, onFailure: (Exception) -> Unit)
```

Comment:
This likely belongs more in the proposal review, but I'll leave it here for now, happy to chat more out of band too.

Should we model this differently? It feels odd to have a suspend fun take in a callback.

Should we just return the result instead?

Reply:
ah it doesn't have to be suspend, just removed it, this method is required for the [googleplay implementation](https://www.tensorflow.org/lite/android/play_services#kotlin_3), which exposed similar Java callbacks